hazelcast/src/main/java/com/hazelcast/cluster/impl/AbstractJoiner.java

```diff
@@ -126,13 +127,14 @@ private void ensureConnectionToAllMembers() {
         if (node.joined()) {
             logger.finest("Waiting for all connections");
             int connectAllWaitSeconds = node.groupProperties.CONNECT_ALL_WAIT_SECONDS.getInteger();
-            int checkCount = 0;
-            while (checkCount++ < connectAllWaitSeconds && !allConnected) {
-                try {
-                    //noinspection BusyWait
-                    Thread.sleep(1000);
-                } catch (InterruptedException ignored) {
-                }
+            ExponentialBackoffSleeper sleeper = ExponentialBackoffSleeper.builder()
+                    .factor(1.3)
+                    .initialMs(50)
+                    .maxMs(1000)
+                    .timeoutMs(TimeUnit.SECONDS.toMillis(connectAllWaitSeconds))
+                    .build();
+            while (sleeper.isTimedOut() && !allConnected) {
```

Comment:
It feels this one is backwards. Shouldn't you break out of the loop is `isTimedOut`?